### PR TITLE
jenkins: Create the `jenkins-cli` command

### PR DESCRIPTION
**Pull request by Pamplemousse**
Signed-off-by: Pamplemousse <xav.maso@gmail.com>

###### Motivation for this change

Supersedes #121756 .

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Pamplemousse** [comment]
CC @nagy

File: pkgs/development/tools/continuous-integration/jenkins/default.nix [hunk 1]
```diff
@@ -10,9 +10,19 @@ stdenv.mkDerivation rec {
...
+
   buildCommand = ''
-    mkdir -p "$out/webapps"
+    mkdir -p "$out/bin" "$out/webapps"
+
     cp "$src" "$out/webapps/jenkins.war"
+
+    # Create the `jenkins-cli` command.
+    ${openjdk}/bin/jar -xf "$src" WEB-INF/lib/cli-${version}.jar \
+      && mv WEB-INF/lib/cli-${version}.jar "$out/bin/jenkins-cli.jar"
```
Comment (bjornfor) [inline pkgs/development/tools/continuous-integration/jenkins/default.nix:22]: Maybe put the jenkins-cli.jar file in `$out/share/java`? From the [nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/):
```
JAR files that are intended to be used by other packages should be installed in $out/share/java. JDKs have a stdenv setup hook that add any JARs in the share/java directories of the build inputs to the CLASSPATH environment variable.
```

I'm not sure who else will be needing this .jar, but it sure feels odd to have it in `$out/bin`.

Comment (bjornfor) [reply pkgs/development/tools/continuous-integration/jenkins/default.nix:22]: The manual also suggests the directory `$out/share/package-name` for private JARs.

**Comment by NeQuissimus** [comment]
Could you add a CLI interaction to the NixOS test?

**Review by fpletz — APPROVED** [review]

**Pull request merged**